packages/adblocker-webextension/src/index.ts

```diff
@@ -405,9 +455,18 @@ export class WebExtensionBlocker extends FiltersEngine {
       typeof TextEncoder !== 'undefined'
     ) {
       const htmlFilters = this.getHtmlFilters(request);
-      if (htmlFilters.length !== 0) {
-        filterRequestHTML(browser.webRequest.filterResponseData, request, htmlFilters);
+      if (htmlFilters.length === 0) {
+        return;
+      }
+      const replaceFilterIndex = htmlFilters.findIndex(([type]) => type === 'replace');
+      // If `replace` filters are not supported
+      if (isRequestHTMLFilterable(request, details) === false && replaceFilterIndex !== -1) {
```

Comment:
how about moving this into `filterRequestHTML`?

Reply:
I think it's making props drilling too deep.